src/lib/y2partitioner/clients/main.rb

```diff
@@ -27,7 +28,10 @@ class Main
       def self.run
         textdomain "storage"
 
+        probed = Y2Storage::StorageManager.instance.y2storage_probed
         staging = Y2Storage::StorageManager.instance.y2storage_staging
+        DeviceGraphs.instance.original = probed
+        DeviceGraphs.instance.current = staging
```

Comment:
Or maybe every single modification is expected to be done withing a `transaction` block? That looks overkill to me.